src/DesignCompile/CompileExpression.cpp

```diff
@@ -488,7 +482,11 @@ constant *compileConst(const FileContent *fC, NodeId child, Serializer &s) {
         if (size.empty()) {
           c->VpiSize(-1);
         } else {
-          c->VpiSize(atoi(size.c_str()));
+          int32_t s = 0;
+          if (NumUtils::parseInt32(size, &s) == nullptr) {
```

Comment:
Instead of comparing with `nullptr`, the code would be more readable using it as bool. The fact that a pointer is returned only is useful for parsing functions that need to advance in a buffer. Using the pointer in expressions where you want to test if a number was parseable is decreasing readability.

So essentially almost everywhere you use `== nullptr` here, use the boolean equivalent. It makes the code _much_ more readable.

```
if (!NumUtils::parseInt(size, &s)) { ... }
```